[docs] update readme

abiogenesis-now opened the pull request:
@hausdorff PTAL, thanks!

seh commented on README.md:
```diff
@@ -3,16 +3,16 @@
 
-Status: Basic functionality works, and the tool is usable.  The focus
-now is on clearer error reporting and advanced features.
+*STATUS: Development is ongoing--this tool is pre-alpha.*
```
- [ ] Nit: Use an em dash there.  
  s/--/—/

seh commented on README.md:
```diff
@@ -3,16 +3,16 @@
+* *Customize* manifests directly with powerful object concatenation syntax
+* *Deploy* app manifests to multiple environments
+* *Diff* across environments to compare two running versions of your app
+* *Version control* the entire state of your app manifests
```
- [ ] "Version control" is odd as a verb.  
  Consider instead a few alternatives:  
  > Track changes to the entire state of your app manifests in your VCS
  > Capture the entire state of your app manifests in your VCS
  > Use your VCS to specify the entire state of your app manifests

seh commented on README.md:
```diff
@@ -3,16 +3,16 @@
 [![Go Report Card](https://goreportcard.com/badge/github.com/ksonnet/ksonnet)](https://goreportcard.com/report/github.com/ksonnet/ksonnet)
 
-A tool for managing Kubernetes resources as code.
+ksonnet provides a framework for you to write, share, and deploy Kubernetes application manifests. With its CLI, you can generate a complete application from scratch in only a few commands, or manage a complex system at scale.
```
- [ ] I stumble over the "for you to" part on each reading.  
  Consider:  
  > _ksonnet_ is a framework for writing, sharing, and deploying ...

jessicayuen commented:
What's here so far looks great.

I'd like to expand on it a little to think through what we want in this README before release. IMO It'd be nice to see the following "sections",

1. Core cases: More or less what's in the README currently for `apply`, `diff`, etc. on files supporting `jsonnet`. 
2. Prototypes (definition & command examples)
    - emphasize on how we are able to quickly create k8s objects
3. Environments (definition & command examples):
    - emphasize on how we are able to manage multiple target servers

abiogenesis-now commented:
thanks @seh for the feedback--I've incorporated and squashed those changes. 

@jessicayuen that's a fair point--I made this PR because @hausdorff wanted an updated blurb, and agree that the current README is not complete as is. That being said, I am in favor of punting any major README changes until we've figured out the plans for the docs website---*unless we have active users of this repo that are not developing the tool*.

^ @hausdorff is that the case?

hausdorff commented:
@abiogenesis-now @jessicayuen Completely agree with Jess, but we should ship this change now so that people stop IM'ing me to complain that it's confusing.

jessicayuen commented:
Ok, that works for me -- just wanted to make sure we capture those cases
before release.

On Oct 24, 2017 5:47 PM, "Jessica Yao" <notifications@github.com> wrote:

thanks @seh <https://github.com/seh> for the feedback--I've incorporated
and squashed those changes.

@jessicayuen <https://github.com/jessicayuen> that's a fair point--I made
this PR because @hausdorff <https://github.com/hausdorff> wanted an updated
blurb, and agree that the current README is not complete as is. That being
said, I am in favor of punting any major README changes until we've figured
out the plans for the docs website---*unless we have active users of this
repo that are not developing the tool*.

^ @hausdorff <https://github.com/hausdorff> is that the case?

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/ksonnet/ksonnet/pull/32#issuecomment-339179122>, or mute
the thread
<https://github.com/notifications/unsubscribe-auth/ADSGxtE0y-NbMtu5hn0tvP6yqm6ZxvV6ks5svoU5gaJpZM4QFRFS>
.
